src/sentry/testutils/cases.py

```diff
@@ -2012,8 +2013,56 @@ class MetricsEnhancedPerformanceTestCase(BaseMetricsLayerTestCase, TestCase):
 
     def setUp(self):
         super().setUp()
+        self.login_as(user=self.user)
         self._index_metric_strings()
 
+    def do_request(self, data: dict[str, Any]) -> Response:
+        """Set up self.features and self.url in the inheriting classes."""
+        with self.feature(self.features):
+            return self.client.get(self.url, data=data, format="json")
+
+    def _on_demand_query_check(
+        self,
+        params: dict[str, Any],
+        groupbys: list[str] | None = None,
+        expected_on_demand_query: bool | None = True,
+        expected_dataset: str | None = "metricsEnhanced",
+    ) -> Response:
+        """Do a request to the events endpoint with metrics enhanced and on-demand enabled."""
+        if params.get("field"):
+            for field in params.get("field"):
+                spec = OnDemandMetricSpec(
+                    field=field,
+                    query=params["query"],
+                    environment=params.get("environment"),
+                    groupbys=groupbys,
+                    spec_type=MetricSpecType.DYNAMIC_QUERY,
+                )
+        elif params.get("yAxis"):
```

Comment:
The rest of this function is the same as it was in the class that inherited this class.

The only addition is this block which supports a table widget test I'm writing on another branch.